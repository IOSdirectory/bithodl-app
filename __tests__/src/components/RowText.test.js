import React from 'react';
import renderer from 'react-test-renderer';
import RowText from '../../../src/components/RowText';

const RealDate = Date;

const mockDate = (fakeDate) => {
  global.Date = class extends RealDate {
    constructor(specifiedDate) {
      super(...arguments);
      return new RealDate(specifiedDate || fakeDate);
    }
  };
};

const unmockDate = () => {
  global.Date = RealDate;
};

describe('RowText', () => {
  beforeAll(() => {
    mockDate('29 November, 2017, 21:59 GMT+02:00');
  });

  afterAll(() => {
    unmockDate();
  });

  it('renders correctly when the item is locked', () => {
    const fakeItem = {
      id: '80049d24-d08e-413f-a35f-d1598ddc8df5',
      hash: '1JNHNJftiWu57UVpp2cjrNNHCeLwbNy3Zt',
      script: 'OP_RETURN',
      amount: 4000000,
      createdAt: 1471651200,
      unlockAt: 1677578920
    };

    const tree = renderer.create(
      <RowText item={fakeItem} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when the item is unlocked', () => {
    const fakeItem = {
      id: 'f05f0af9-66c5-4d76-9882-3bb2377aa863',
      hash: '1JNHNJftiWu57UVpp2cjrNNHCeLwbNy3Zt',
      script: 'OP_RETURN',
      amount: 2000000,
      createdAt: 1471651200,
      unlockAt: 1482192000
    };

    const tree = renderer.create(
      <RowText item={fakeItem} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
