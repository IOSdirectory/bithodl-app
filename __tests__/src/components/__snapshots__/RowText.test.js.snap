// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowText renders correctly when the item is locked 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  numberOfLines={undefined}
  style={
    Array [
      Object {
        "color": "#504F63",
        "fontFamily": "Avenir Next",
        "fontWeight": "400",
      },
      Object {
        "color": "#8F8E94",
        "fontSize": 13,
        "fontWeight": "400",
        "lineHeight": 18,
      },
    ]
  }
>
  Until
   
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#504F63",
          "fontFamily": "Avenir Next",
          "fontWeight": "400",
        },
        Object {
          "color": "#8F8E94",
          "fontSize": 13,
          "fontWeight": "400",
          "lineHeight": 18,
        },
      ]
    }
  >
    28 Feb, 2023 11:08
  </Text>
</Text>
`;

exports[`RowText renders correctly when the item is unlocked 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  numberOfLines={undefined}
  style={
    Array [
      Object {
        "color": "#504F63",
        "fontFamily": "Avenir Next",
        "fontWeight": "400",
      },
      Object {
        "color": "#8F8E94",
        "fontSize": 13,
        "fontWeight": "400",
        "lineHeight": 18,
      },
    ]
  }
>
  Ended
   
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#504F63",
          "fontFamily": "Avenir Next",
          "fontWeight": "400",
        },
        Object {
          "color": "#8F8E94",
          "fontSize": 13,
          "fontWeight": "400",
          "lineHeight": 18,
        },
      ]
    }
  >
    20 Dec, 2016 01:00
  </Text>
</Text>
`;
